components/kyma-environment-broker/internal/broker/instance_update.go

```diff
@@ -348,3 +358,31 @@ func (b *UpdateEndpoint) isKyma2(instance *internal.Instance) (bool, string, err
 	kv := s.GetKymaVersion()
 	return internal.DetermineMajorVersion(kv) == 2, kv, nil
 }
+
+// processExpirationParam returns true, if expired
+func (b *UpdateEndpoint) processExpirationParam(instance *internal.Instance, details domain.UpdateDetails) (*internal.Instance, error) {
+	// if the instance was expired before (in the past), then no need to do any work
+	if instance.IsExpired() {
```

Comment:
it's worth keeping this to be on the safe side but I believe this is not reachable because KEB update endpoint rejects any update requests earlier in the call stack with 422
https://github.com/kyma-project/control-plane/blob/445ba9036db31730f1073ae1f6524e606b8c0f1d/components/kyma-environment-broker/internal/broker/instance_update.go#L93-L96